### ISSUE TITLE
Shall responder skip the mutual authentication if requester does not have the capability?

### ISSUE THREAD

**Issue by gaojiaqi7** [DMTF/libspdm#2121]
In the implementation of libspdm, responder sets the value of `mut_auth_requested` of the key exchange response according to the capability of requester and responder:
https://github.com/DMTF/libspdm/blob/04baaa92bc59e03aa528988edd46bfc15ce01e2b/library/spdm_responder_lib/libspdm_rsp_key_exchange.c#L386-L399
If responder or requester does not have such capability, responder will skip the mutual authentication and run the rest of the process and finally establish the secure session, which may not be safe.

I tried the spdm-emu to run a responder with argument `--mut_auth DIGESTS` and a requester without `MUT_AUTH` capability using below commands. The result shows they skip the mutual authentication and successfully running to the end.
```
./spdm_responder_emu --trans PCI_DOE --mut_auth DIGESTS --req_asym ECDSA_P384 --basic_mut_auth NO

./spdm_requester_emu --trans PCI_DOE --req_asym ECDSA_P384 --exe_conn DIGEST,CERT,CHAL --exe_session KEY_EX,PSK,KEY_UPDATE,HEARTBEAT,DIGEST,CERT --cap CERT,CHAL,ENCRYPT,MAC,KEY_EX,PSK,HBEAT,KEY_UPD,HANDSHAKE_IN_CLEAR
```

My thought is that responder should end the conversation and return an error message to requester if responder **wants to verify the requester** but requester does not support the `mut_auth` capability.

**Comment by jyao1** [comment]
Good point.
I think we need to distinguish capability from enabling.
* If Responder just wants to support capability, then it should allow the requester not supporting mut_auth.
* If responder wants to enforce the mut_auth enabling state, then it should reject the requester not supporting mut_auth.

**Comment by steven-bellock** [comment]
Right, the Integrator may need to be in the loop to determine if mutual authentication is needed. In particular, `MutAuthRequested` may be conditional based on the Requester's `OpaqueData`. There may also be a case where the Responder requires mutual authentication, but it's just for logging purposes and does not restrict what the Requester can do.
